docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,651 @@
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example updates an existing issue's status and adds a comment with deployment details.
+
+```yaml
+steps:
+  - uses: jira
+    config:
+      credentials:
+        secretName: jira-credentials
+      updateIssue:
+        issueKey: DEPLOY-123
+        status: "IN PROGRESS"
+        summary: "Deploy ${{ imageFrom(vars.imageRepo).Tag }} to ${{ ctx.stage.name }} - IN PROGRESS"
+        addLabels:
+          - deploying
+          - "${{ ctx.stage.name }}-deployment"
+        customFields:
+          customfield_10000: "${{ ctx.stage.name }} Environment"
+          customfield_10001: "${{ ctx.promotion.name }}"
+```
+
+### Delete Issue
+
+Deletes a Jira issue and optionally its subtasks.
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                    |
+| ---------------------------- | --------- | -------- | ---------------------------------------------- |
+| `deleteIssue.issueKey`       | `string`  | Y        | The Jira Issue Key (e.g., EXT-123).            |
+| `deleteIssue.deleteSubtasks` | `boolean` | N        | If true, all subtasks will be deleted as well. |
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example deletes a Jira issue and all its subtasks when a deployment is rolled back.
+
+```yaml
+steps:
+  - uses: jira
+    config:
+      credentials:
+        secretName: jira-credentials
+      deleteIssue:
+        issueKey: "${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}"
+        deleteSubtasks: true
+```
+
+### Search Issues
+
+Searches for Jira issues using JQL (Jira Query Language).
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                                                                   |
+| ---------------------------- | --------- | -------- | --------------------------------------------------------------------------------------------- |
+| `searchIssue.jql`            | `string`  | Y        | The JQL query to search for issues.                                                           |
+| `searchIssue.expectMultiple` | `boolean` | N        | If true, expects multiple results. If false, expects single result and fails with >1 results. |
+| `searchIssue.fields`         | `array`   | N        | List of fields to include in search results.                                                  |
+| `searchIssue.expands`        | `array`   | N        | List of fields to expand in search results.                                                   |
+
+#### Output
+
+| Name    | Type     | Description                                                                 |
+| ------- | -------- | --------------------------------------------------------------------------- |
+| `issue` | `object` | The found Jira issue object containing all requested fields and expansions. |
```

Comment:
It's not clear from this description what happens if `expectMultiple` was `true` and multiples were found. It leaves me guessing.

Reply:
Much clearer now!